src/analyses/cfg_dominators.h

```diff
@@ -115,6 +246,26 @@ class cfg_dominators_templatet
 protected:
   void initialise(P &program);
   void fixedpoint(P &program);
+
+  /// Goal of this is to assign post-order numbering to each node in the tree.
+  /// Full explanation here: https://en.wikipedia.org/wiki/Tree_traversal.
+  void assign_postordering(std::size_t start_node_index);
+
+  /// Sort our CFG nodes in reverse post-order. Just gets every node that we've
+  /// processed (anything not NODE_NOT_VISITED) then order from highest to
+  /// lowest. This excludes the entry-node though.
+  std::vector<T>
+  get_reverse_postordered_instructions(std::size_t entry_node_index) const;
+
+  /// Walks up the dominators of left/right nodes until it finds one that is
+  /// common to both sides. Used to work out a the common immediate dominator
```

Comment:
"a the common immediate dominator" -> "the least common dominator" (i.e. the dominator that is common to both, and which dominates no other common dominator)

Reply:
Is 'least common denominator' a known phrase? It reads really awkwardly otherwise.

Reply:
Looks like the go-to phrase is in fact "nearest common dominator"